src/ribbon/art_aui.cpp

```diff
@@ -252,11 +252,18 @@ void wxRibbonAUIArtProvider::SetColourScheme(
 #ifdef __WXOSX__
     m_tab_label_colour = wxSystemSettings::GetColour(wxSYS_COLOUR_CAPTIONTEXT);
 #else
-    m_tab_label_colour = LikePrimary(0.1);
+    if (wxSystemSettings::GetAppearance().IsDark())
+    {
+        m_tab_label_colour = wxSystemSettings::GetColour(wxSYS_COLOUR_CAPTIONTEXT);
+    }
+    else
+    {
+        m_tab_label_colour = LikePrimary(0.1);
+    }
```

Comment:
This can be written more briefly and, IMHO, nicely as

```suggestion
    m_tab_label_colour = wxSystemSettings::SelectLightDark()
        ? LikePrimary(0.1)
        : wxSystemSettings::GetColour(wxSYS_COLOUR_CAPTIONTEXT);
```